### DISCUSSION TITLE
Pelvis Smoothed Surface from DICOM Image Segmentation

### DISCUSSION THREAD

**Discussion by miritrope** [pyvista/pyvista#4952] [Q&A]
Hello everyone,

I'm interested in generating a smoothed and closed surface representation of the female pelvis (body) using an MRI DICOM image. It doesn't need to be exact, just smooth enough for presentation purposes. Here are the general steps I've taken so far:

1. I used SimpleITK (SITK) to iterate through the image to find the first non-zero voxel. Here's an example of the code I used for this step:

```python
# Iterate through the image to find the first non-zero voxel
for i in range(thresh_image.GetWidth()):
    for j in range(thresh_image.GetHeight()):
        for k in range(thresh_image.GetDepth()):
            intensity = thresh_image.GetPixel(i, j, k)

            # Check if the voxel corresponds to an object (not background)
            if intensity:
                # Set the corresponding voxel in the segmentation image to 255 (white)
                segmentation_image.SetPixel(i, j, k, 1)
                break

# Repeat the same process in reverse to ensure completeness
for i in range(thresh_image.GetWidth() - 1, -1, -1):
    for j in range(thresh_image.GetHeight() - 1, -1, -1):
        for k in range(thresh_image.GetDepth() - 1, -1, -1):
            intensity = thresh_image.GetPixel(i, j, k)

            # Check if the voxel corresponds to an object (not background)
            if intensity:
                # Set the corresponding voxel in the segmentation image to 255 (white)
                segmentation_image.SetPixel(i, j, k, 1)
                break
```

2. I saved the resulting segmentation image as an NII file (attached).
[segmentation_image.nii.zip](https://github.com/pyvista/pyvista/files/12724705/segmentation_image.nii.zip)

3. I opened the NII file in MITK and used the "create polygon model" function, saving the model as a .VTK file (attached).
[Uploading segmentation_image.vtk.zip…]()

However, I'm now looking for ways to further improve this output. Specifically, I want to achieve a smoothed and closed surface without any holes. I've attempted to use various filters in ParaView, such as Delaunay3D, Decimate, Smooth, Curvature, but so far, I haven't had much success. This is the mesh (surface output)

<img width="933" alt="Screenshot 2023-09-26 at 10 50 18" src="https://github.com/pyvista/pyvista/assets/47087140/e6650086-cd0c-4f95-9dcf-1acc2838220f">

Interestingly, I did manage to create a beautiful surface using this algorithm for the head (scalp or skull segmentation), represented in red in the attached image.

<img width="853" alt="Screenshot 2023-09-22 at 16 54 07" src="https://github.com/pyvista/pyvista/assets/47087140/ec37ffcc-c69d-4519-b321-09ab2a8ffa64">

If anyone has suggestions or recommendations on how to achieve a smoother and closed surface representation from the output I have, I would greatly appreciate your input.

Thank you!

**Comment by banesullivan** [comment]
This would be a great topic to reach out to the fantastic people at Kitware about (ref #4033)
